src/Concordium/Client/Runner.hs

```diff
@@ -570,13 +592,16 @@ processTransactionCmd action baseCfgDir verbose backend =
         when (ioTail intOpts) $ do
           tailTransaction_ hash
 --          logSuccess [ "credential deployed successfully" ]
-    TransactionStatus h -> do
+
+    TransactionStatus h schemaFile -> do
       hash <- case parseTransactionHash h of
         Nothing -> logFatal [printf "invalid transaction hash '%s'" h]
         Just hash -> return hash
-      status <- withClient backend $ queryTransactionStatus hash
-      when verbose $ logInfo [printf "Response: %s" (show status)]
-      runPrinter $ printTransactionStatus status True
+      withClient backend . withBestBlockHash Nothing $ \bb -> do
+        status <- queryTransactionStatus hash
+        schema <- getEventSchema verbose schemaFile status $ Just bb
+        runPrinter $ printTransactionStatus status True schema
```

Comment:
The schema should not be from the best block. It should be from the block that the transaction status returns.

The transaction status is a mapping from block hashes to outcomes. We need to get the relevant schema out.

While using the "best" block will almost always be correct, it will not **always** be correct.